source/mir/sum.d

```diff
@@ -1838,22 +1838,31 @@ nothrow unittest
         foreach (t; test[0]) summator.put(t);
         auto r = test[1];
         auto s = summator.sum;
-        version(X86)
+        import core.exception: AssertError;
+        try
         {
-            import std.math: nextDown, nextUp;
-            assert(summator.isNaN() == r.isNaN());
-            assert(summator.isFinite() == r.isFinite() || r == -double.max
-                   && s == -double.infinity || r == double.max && s == double.infinity);
-            assert(summator.isInfinity() == r.isInfinity() || r == -double.max
-                   && s == -double.infinity || r == double.max && s == double.infinity);
-            assert(nextDown(s) <= r && r <= nextUp(s) || s.isNaN && r.isNaN);
+            version(X86)
+            {
+                import std.math: nextDown, nextUp;
+                assert(summator.isNaN() == r.isNaN());
+                assert(summator.isFinite() == r.isFinite() || r == -double.max
+                       && s == -double.infinity || r == double.max && s == double.infinity);
+                assert(summator.isInfinity() == r.isInfinity() || r == -double.max
+                       && s == -double.infinity || r == double.max && s == double.infinity);
+                assert(nextDown(s) <= r && r <= nextUp(s) || s.isNaN && r.isNaN);
+            }
+            else
+            {
+                assert(summator.isNaN() == r.isNaN());
+                assert(summator.isFinite() == r.isFinite());
+                assert(summator.isInfinity() == r.isInfinity());
+                assert(s == r || s.isNaN && r.isNaN);
+            }
         }
-        else
+        catch(AssertError e)
         {
-            assert(summator.isNaN() == r.isNaN());
-            assert(summator.isFinite() == r.isFinite());
-            assert(summator.isInfinity() == r.isInfinity());
-            assert(s == r || s.isNaN && r.isNaN);
+            import std.stdio;
+            stderr.writefln("i = %s, result = %s (%X), target = %s (%X)", i, s, s, r, r);
         }
```

Comment:
Then go for it, but don't forget to remove it afterwards cause currently the test would pass with this error 
;-)

On April 25, 2016 11:01:49 AM GMT+03:00, Ilya Yaroshenko <notifications@github.com> wrote:
>>          {
>> -            assert(summator.isNaN() == r.isNaN());
>> -            assert(summator.isFinite() == r.isFinite());
>> -            assert(summator.isInfinity() == r.isInfinity());
>> -            assert(s == r || s.isNaN && r.isNaN);
>> +            import std.stdio;
>> +            stderr.writefln("i = %s, result = %s (%X), target = %s
>(%X)", i, s, s, r, r);
>>          }
>
>> debug something specific
>
>Yes
>
>---
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/libmir/mir/pull/197/files/b763f27a216a866598a8f008e6dd07ee42e3ecd9#r60874181